core/src/main/java/org/elasticsearch/search/sort/GeoDistanceSortBuilder.java

```diff
@@ -22,41 +22,109 @@
 import org.elasticsearch.ElasticsearchParseException;
 import org.elasticsearch.common.geo.GeoDistance;
 import org.elasticsearch.common.geo.GeoPoint;
+import org.elasticsearch.common.geo.GeoUtils;
+import org.elasticsearch.common.io.stream.NamedWriteable;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.unit.DistanceUnit;
+import org.elasticsearch.common.xcontent.ToXContent;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
 import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryParseContext;
+import org.elasticsearch.search.MultiValueMode;
 
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Locale;
+import java.util.Objects;
 
 /**
  * A geo distance based sorting on a geo point like field.
  */
-public class GeoDistanceSortBuilder extends SortBuilder {
+public class GeoDistanceSortBuilder extends SortBuilder implements ToXContent, NamedWriteable<GeoDistanceSortBuilder>, ParameterParser<GeoDistanceSortBuilder> {
+    public static final String NAME = "_geo_distance";
+    public static final boolean DEFAULT_COERCE = false;
+    public static final boolean DEFAULT_IGNORE_MALFORMED = false;
+    public static final String DEFAULT_SORT_MODE = "MAX";
 
-    final String fieldName;
+    static final GeoDistanceSortBuilder PROTOTYPE = new GeoDistanceSortBuilder("", -1, -1);
+
+    private final String fieldName;
     private final List<GeoPoint> points = new ArrayList<>();
-    private final List<String> geohashes = new ArrayList<>();
 
-    private GeoDistance geoDistance;
-    private DistanceUnit unit;
-    private SortOrder order;
-    private String sortMode;
+    private GeoDistance geoDistance = GeoDistance.DEFAULT;
+    private DistanceUnit unit = DistanceUnit.DEFAULT;
+    private SortOrder order = SortOrder.DEFAULT;
```

Comment:
Can you re-check the default? I think it should be "ASC" since GeoDistanceSortParser used to set `reverse = false` by default to far, which I think translates to ASC?